src/database.py

```diff
@@ -1,74 +1,112 @@
-import os
+from client import client
+from constants import (
+    DATA,
+    UNITS,
+    UNIT_ID,
+    SCHEMAS,
+    DATASETS,
+    VERSIONS,
+    SURVEY_ID,
+    DATASET_ID,
+    DOUBLE_EQUALS,
+    LATEST_VERSION,
+    DATASET_SCHEMAS,
+    DATASET_SCHEMA_ID
+)
 
-import firebase_admin
-from firebase_admin import firestore
 
-cred_obj = firebase_admin.credentials.Certificate(
-    os.environ.get("FIREBASE_KEYFILE_LOCATION")
-)
-default_app = firebase_admin.initialize_app(cred_obj)
-db = firestore.client()
-datasets_collection = db.collection("datasets")
-schemas_collection = db.collection("schemas")
+schemas_collection = client.collection(SCHEMAS)
+
+datasets_collection = client.collection(DATASETS)
 
 
 def set_dataset(dataset_id, dataset):
-    dataset.pop("data")
+    dataset.pop(DATA)
+
     datasets_collection.document(dataset_id).set(dataset)
 
 
 def set_data(dataset_id, data):
-    units_collection = datasets_collection.document(dataset_id).collection("units")
-    units_collection.document(data["unit_id"]).set(data)
+    units_collection = datasets_collection.document(dataset_id).collection(UNITS)
+
+    units_collection.document(data[UNIT_ID]).set(data)
 
 
 def get_data(dataset_id, unit_id):
-    units_collection = datasets_collection.document(dataset_id).collection("units")
+    units_collection = datasets_collection.document(dataset_id).collection(UNITS)
+
     return units_collection.document(unit_id).get().to_dict()
 
 
 def set_schema(dataset_schema_id, survey_id, dataset_schema):
     dataset_schema_versions = schemas_collection.document(dataset_schema_id)
+
     if not dataset_schema_versions.get().exists:
-        dataset_schema_versions.set({"latest_version": 1, "survey_id": survey_id})
+        dataset_schema_versions.set({LATEST_VERSION: 1, SURVEY_ID: survey_id})
```

Comment:
With moving everything to constants, it feels as if we will the human readable element of what we are trying to achieve is attracted away, however what might be better is to have a class model which will map to these constants to give them context. Happy to discuss this further.